src/main/kotlin/org/rust/cargo/util/CargoArgsSeparator.kt

```diff
@@ -0,0 +1,123 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.cargo.util
+
+class CargoRunArgsSeparator(cargoArgs: List<String>) : AbstractCargoArgsSeparator(cargoArgs, RUN_OPTIONS) {
+    override val commandOptions: List<String> by lazy {
+        separatedArgs.first
+    }
+
+    override val executableArguments: List<String> by lazy {
+        val positionalArguments = separatedArgs.second
+        if (positionalArguments.isNotEmpty() && positionalArguments.first() == "--") {
+            positionalArguments.drop(1)
+        } else {
+            positionalArguments
+        }
```

Comment:
Why can we drop `--` while calculation of `separatedArgs`?

Reply:
Cargo does it too.

```rust
fn main() {
    use std::env;
    let args: Vec<String> = env::args().collect();
    println!("{:?}", args);
}
```

```
cargo run a -- -- b
["target/debug/rust-sandbox", "a", "--", "--", "b"]
```

```
cargo run -- -- b
["target/debug/rust-sandbox", "--", "b"]
```